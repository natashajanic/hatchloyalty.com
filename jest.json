{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
