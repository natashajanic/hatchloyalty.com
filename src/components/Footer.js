import React from 'react'
import system from 'system-components'

const UIFooter = system({
  is: 'footer',
  alignItems: 'center',
  bg: 'white',
  display: 'flex',
  justifyContent: 'space-between',
  mx: 'auto',
  px: 3,
  py: 3,
  width: '1024px',
})


const Footer = () => (
  <UIFooter>
    Footer
  </UIFooter>
)


export default Footer
