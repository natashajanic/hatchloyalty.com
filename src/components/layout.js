import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { StaticQuery, graphql } from 'gatsby'
import theme from '../theme'
import system from 'system-components'
import { ThemeProvider } from 'styled-components'
import Header from './header'
import Footer from './Footer'
import './layout.css'

const PageWrapper = system({
  is: 'div',
  bg: 'white',
},
  'bg',
  'color',
)

const Layout = ({ children, pageStyle }) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => (
      <>
        <Helmet
          title={data.site.siteMetadata.title}
          meta={[
            { name: 'description', content: 'Hatch: Loyalty is an outcome.' },
            { name: 'keywords', content: 'hatch, loyalty, hatch loyalty,' },
          ]}
        >
          <html lang="en" />
        </Helmet>

        <ThemeProvider theme={theme}>
          <PageWrapper bg={pageStyle}>
            <Header siteTitle={data.site.siteMetadata.title} />
            {children}
            <Footer />
          </PageWrapper>
        </ThemeProvider>
      </>
    )}
  />
)

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
