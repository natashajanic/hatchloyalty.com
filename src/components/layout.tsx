import * as autoBind from 'auto-bind'
import { StaticQuery, graphql } from 'gatsby'
import * as React from 'react'
import Helmet from 'react-helmet'
import system from 'system-components'
import { ThemeProvider } from 'styled-components'
import theme from 'src/theme'
import Header from './header'
import Footer from './footer'

/* tslint:disable-next-line no-import-side-effect */
import './layout.css'

const PageWrapper = system(
  {
    is: 'div',
    bg: 'white',
  },
  'bg',
  'color'
)

const Main = system({
  is: 'main',
})

interface ILayoutProps {
  pageStyle?: any
}

class Layout extends React.Component<ILayoutProps, {}> {
  constructor(props: ILayoutProps) {
    super(props)
    autoBind(this)
  }

  renderContent(data: any) {
    const { children, pageStyle } = this.props
    const meta = [
      {
        name: 'description',
        content: 'Hatch: Loyalty is an outcome.',
      },
      {
        name: 'keywords',
        content:
          'hatch, loyalty, hatch loyalty, api, customer activation, data activation, platform',
      },
    ]
    return (
      <>
        <Helmet title={data.site.siteMetadata.title} meta={meta}>
          <html lang="en" />
        </Helmet>

        <ThemeProvider theme={theme}>
          <PageWrapper bg={pageStyle}>
            <Header
              navLinks={data.site.siteMetadata.navLinks}
              siteTitle={data.site.siteMetadata.title}
            />
            <Main role="main">{children}</Main>
            <Footer />
          </PageWrapper>
        </ThemeProvider>
      </>
    )
  }

  render() {
    const query = graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
            navLinks {
              name
              link
            }
          }
        }
      }
    `

    return <StaticQuery query={query} render={this.renderContent} />
  }
}

export default Layout
