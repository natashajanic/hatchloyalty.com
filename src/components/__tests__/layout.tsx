import { shallow, ShallowWrapper } from 'enzyme'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Helmet from 'react-helmet'
import Footer from '../footer'
import Header from '../header'
import { IPureLayoutProps, PureLayout } from '../layout'

describe('Layout', () => {
  let props: IPureLayoutProps
  let mountedComponent: ShallowWrapper<any, any, any>

  beforeEach(() => {
    props = {
      pageStyle: 'offWhite',
      data: {
        site: {
          siteMetadata: {
            title: 'Hatch Loyalty',
            navLinks: [
              {
                name: 'Home',
                link: '/',
              },
              {
                name: 'Resources',
                link: '/resources',
              },
              {
                name: 'Blog',
                link: '/blogs',
              },
            ],
          },
        },
      },
    }
    mountedComponent = shallow(
      <PureLayout {...props}>Child Content</PureLayout>
    )
  })

  describe('render', () => {
    it('renders a helmet', async () => {
      expect(mountedComponent.find(Helmet).length).toEqual(1)
      const helmet = mountedComponent.find(Helmet).at(0)
      expect(helmet.prop('title')).toEqual(props.data.site.siteMetadata.title)
    })

    it('renders a header', async () => {
      expect(mountedComponent.find(Header).length).toEqual(1)
    })

    it('renders the contained content', async () => {
      expect(mountedComponent.find('[role="main"]').length).toEqual(1)
      expect(
        mountedComponent
          .find('[role="main"]')
          .at(0)
          .text()
      ).toContain('Child Content')
    })

    it('renders a footer', async () => {
      expect(mountedComponent.find(Footer).length).toEqual(1)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
