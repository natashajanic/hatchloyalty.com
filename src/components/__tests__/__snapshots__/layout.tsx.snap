// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render matches the expected snapshot 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    meta={
      Array [
        Object {
          "content": "Hatch: Loyalty is an outcome.",
          "name": "description",
        },
        Object {
          "content": "hatch, loyalty, hatch loyalty, api, customer activation, data activation, platform",
          "name": "keywords",
        },
      ]
    }
    title="Hatch Loyalty"
  >
    <html
      lang="en"
    />
  </HelmetWrapper>
  <ThemeProvider
    theme={
      Object {
        "borders": Array [
          0,
          "1px solid",
          "2px solid",
        ],
        "breakpoints": Array [
          "960px",
          "1200px",
          "1400px",
        ],
        "colorStyles": Object {
          "completedStep": Object {
            "backgroundColor": "#63bab0",
            "borderColor": "#63bab0",
            "color": "#ffffff",
          },
          "currentStep": Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#63bab0",
            "color": "#63bab0",
            "fontWeight": 600,
          },
          "danger": Object {
            "backgroundColor": "#f7e2e2",
            "color": "#9c292d",
          },
          "default": Object {
            "backgroundColor": "#f7f7f7",
            "border": "1px solid #dfdfdf",
            "color": "#666666",
          },
          "defaultAlt": Object {
            "backgroundColor": "#ffffff",
            "color": "#666666",
          },
          "disabledStep": Object {
            "borderColor": "#dfdfdf",
            "color": "#dfdfdf",
            "cursor": "not-allowed",
          },
          "futureStep": Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#dfdfdf",
            "color": "#666666",
          },
          "info": Object {
            "backgroundColor": "#CDECFF",
            "color": "#2d5d95",
          },
          "primary": Object {
            "backgroundColor": "#ebfcfa",
            "color": "#04575b",
          },
          "secondary": Object {
            "backgroundColor": "#ebfcfa",
            "color": "#04575b",
          },
          "warning": Object {
            "backgroundColor": "#fceddf",
            "color": "#594430",
          },
        },
        "colors": Object {
          "black": "#232323",
          "blue": "#3ca1df",
          "blueDark": "#2d5d95",
          "blueLight": "#CDECFF",
          "gray": "#999999",
          "grayDark": "#666666",
          "grayLight": "#dfdfdf",
          "green": "#63bab0",
          "greenDark": "#04575b",
          "greenLight": "#79e9dd",
          "greenXLight": "#ebfcfa",
          "offBlack": "#484848",
          "offWhite": "#f7f7f7",
          "red": "#c34848",
          "redDark": "#9c292d",
          "redLight": "#f7e2e2",
          "white": "#ffffff",
          "yellow": "#ffc58b",
          "yellowDark": "#594430",
          "yellowLight": "#fceddf",
        },
        "fontSizes": Array [
          10,
          12,
          14,
          16,
          18,
          20,
          24,
          30,
          36,
          48,
          60,
          72,
          96,
        ],
        "fontWeights": Object {
          "bold": 600,
          "light": 300,
          "normal": 400,
        },
        "lineHeight": "1.5",
        "listTypes": Array [
          "none",
          "disc",
        ],
        "shadows": Array [
          "none",
          "inset 0 0 0 1px #dfdfdf",
          "inset 0 0 0 1px #dfdfdf, 0 0 4px #999999",
          "0 2px 6px 0 #dfdfdf",
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
        ],
        "textColor": "#484848",
        "textStyles": Object {
          "lowercase": Object {
            "textTransform": "lowercase",
          },
          "normal": Object {
            "textTransform": "none",
          },
          "uppercase": Object {
            "letterSpacing": "0.1em",
            "textTransform": "uppercase",
          },
        },
      }
    }
  >
    <Styled(Component)
      bg="offWhite"
      blacklist={
        Array [
          "css",
          "color",
          "bg",
        ]
      }
      is="div"
    >
      <Header
        navLinks={
          Array [
            Object {
              "link": "/",
              "name": "Home",
            },
            Object {
              "link": "/resources",
              "name": "Resources",
            },
            Object {
              "link": "/blogs",
              "name": "Blog",
            },
          ]
        }
        siteTitle="Hatch Loyalty"
      />
      <Styled(Component)
        blacklist={
          Array [
            "css",
          ]
        }
        is="main"
        role="main"
      >
        Child Content
      </Styled(Component)>
      <Footer />
    </Styled(Component)>
  </ThemeProvider>
</Fragment>
`;
